src/game_battlealgorithm.cpp

```diff
@@ -713,6 +717,25 @@ int Game_BattleAlgorithm::Normal::GetSourcePose() const {
 		: lcf::rpg::BattlerAnimation::Pose_AttackRight;
 }
 
+int Game_BattleAlgorithm::Normal::GetCBAMovement() const {
```

Comment:
How does this work when dual wielding and the second weapon is used?

What if the the first weapon causes you to walk to the enemy and the second one doesn't? How about the opposite case?

Reply:
I must admit here that what I "built" here doesn't exactly match RPG_RT (hence the accuracy of only about 80%). What is different here is that we have separate battle algorithms for every attack. This means that on every attack a move is started (depending on the weapon's settings), then the battle animation is played on the target and then the move back occurs. The effects of the attack are applied afterwards. Sadly I had trouble to get it exactly the way RPG_RT does. In RPG_RT it works this way: First the actor starts the move, then make one or more attacks on the targets where the effects are applied after the attack animation and then moves back. In the case of moving to the target the actor remains at the first target even if it gets killed and the actor attacks the next target. The problem I had with the RPG_RT way is when a victory condition was triggered, then the attacking actor didn't move back to his starting point and trying to work around the victory condition led to even more bad code in my case. The version I built has the advantage that the actor is always in the right position when attacking a target, but the disadvantage is it looks awkward when attacking the same target two or more times in a row (always moving back and forth).

Reply:
I'm all for an 80% solution being better than the 0% we have now, however I do think having the actor go to and back for every hit will greatly slow the pace of battles. We should try to fix that.

I would be happy with fixing it correctly, or failing that have the actor only walk to the first target and stay there until the battle sequence is done.

Why can't you do the "move back" before `PostAction` and after the repeated algorithm executions. Only one battle algo is used, its' just replayed multiple times for each hit.

Basically here in `ProcessBattleActionApply` you would inject your MoveBack before PostAction

```
// If action does multiple attacks, repeat again.
if (action->RepeatNext(false)) {
    SetBattleActionState(BattleActionState_StartAlgo);
    return BattleActionReturn::eContinue;
}

SetBattleActionState(BattleActionState_PostAction);
return BattleActionReturn::eContinue;
```

The way I think this would work:

* Start - setup the algo
* MoveToTarget - move to the target from whatever your current position is, or don't move if you're already at the target. This will work seamlessly for multiple hits against the same target
* Animations, Reflect, execute, blah blah..
* Apply - apply effects. Check if we need to do another hit, if so, go back to Start state
* MoveBackToPosition
* PostAction ... Finished

Reply:
To answer your first comment: In case of dual wielding the movement information of the first weapon is used.

Reply:
You mean that's how RPG_RT does it? Always the first weapon?

We can double check. Equip 2 weapons. First one moves, second one does not. First attack kills enemy. Does the second attack move to the second enemy or stay put?

If that's the case we should change our code to reflect that + a comment describing why.

Reply:
I tested this one. When the first enemy gets killed the actor remains at the place from the first enemy when attacking the second enemy. After that he returns to his starting point. So the information of the first weapon is used. The code already works.